### PR TITLE
Use separate sets of object files for static and shared libs

### PR DESCRIPTION
**Pull request by igelhaus**
* Static library is built from non-PIC object files.
* Shared library is built from PIC object files.
* Corresponding changes are propagated to all dependencies.
* Updated build documentation.
* Introduced fixes in building `udis86` and Intel VTune JIT helper.

### CODE REVIEW

**Comment by igelhaus** [comment]
@eliasdaler Please review.

For historical reasons, LuaVela's static library was built from PIC object files. This patch fixes this issue.

File: INSTALL.rst [hunk 1]
```diff
@@ -54,24 +54,32 @@ dependencies, which are not built by default.
 Building
 --------
 
-To trigger the build with default configuration, enter
+Both in source and out of source builds are supported, the latter is
+recommended. To trigger build with default configuration, please run:
 
- $ cmake . [...configuration flags...]
- $ make
+ $ make luavela-build
```
[the remark this answers is not in the document]
Comment (igelhaus) [reply INSTALL.rst]: Fixed, thanks.

File: INSTALL.rst [hunk 2]
```diff
@@ -92,11 +100,13 @@ CMAKE_INSTALL_PREFIX
...
-want something like this for packaging and production installs:
+resorting to the value provided by CMake. However, you almost definitely
+want something like this for production installs:
+
+ $ cmake /path/to/luavela -DCMAKE_INSTALL_PREFIX=/usr
+ $ sudo make install
 
- $ cmake . -DCMAKE_INSTALL_PREFIX=/usr
- $ make package
+If you have a CPack file, `make package` should be used for creating a package.
```
[the remark this answers is not in the document]
Comment (igelhaus) [reply INSTALL.rst]: Fixed, thanks.

File: INSTALL.rst [hunk 3]
```diff
@@ -54,24 +54,32 @@ dependencies, which are not built by default.
...
+Libraries and binaries are created in `luavela-build/src`.
+
+All CMake configuration flags should be passed in -DKEY=VALUE form. If KEY is a
+boolean flag, any CMake-compatible boolean value can be used as VALUE (i.e.
 1/0, ON/OFF, TRUE/FALSE). "ON" or "OFF" are used throughout this documentation
 for consistency. Example of triggering build with some flags defined:
 
- $ cmake . -DCMAKE_BUILD_TYPE=Debug
- $ make VERBOSE=1
+ $ make luavela-build
```
[the remark this answers is not in the document]
Comment (igelhaus) [reply INSTALL.rst]: Fixed, thanks.

**Comment by igelhaus** [comment]
@eliasdaler @igormunkin Fixed all remarks.

**Comment by igelhaus** [comment]
FYI issues with CI will be fixed as soon as #26 is merged.

**Comment by eliasdaler** [comment]
Merged here manually: https://github.com/iponweb/luavela/commit/f7591786c8ed3562361fe55b5b337bf17fd4a8c3

**Pull request closed without merging**